org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/editor/syntaxcoloring/SemanticHighlightingStyleToTokenMapper.xtend

```diff
@@ -0,0 +1,22 @@
+/*******************************************************************************
+ * Copyright (c) 2018 TypeFox and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.tests.testlanguage.editor.syntaxcoloring
+
+import org.eclipse.xtext.ide.server.semanticHighlight.ISemanticHighlightingStyleToTokenMapper
+
+class SemanticHighlightingStyleToTokenMapper implements ISemanticHighlightingStyleToTokenMapper {
+
+	override toScopes(String styleId) {
+		return #[styleId];
+	}
+
+	override getAllStyleIds() {
```

Comment:
Do you have plans to work on a different impl of getAllStyleIds, e.g. similar to how we  collect the style ids in the eclipse case?

Reply:
No, not in the scope of this PR. As I see, even if we would come up with an adapter layer (to collect all the style IDs just like in the Eclipse-case), the actual implementations are in the UI code for the downstream projects. So it would be useless. But, if you have already something in mind, a silver-bullet solution, I am not against implementing it, so please share your idea.

Reply:
CC: @meysholdt

Reply:
My main concern is not about the reuse of some piece of code, but the general structure of the API. A method `getAllStyleIds` that has to be in sync with the style ids anyways, is prone to be forgotten to override. I'd like to reduce the risk and the burden on the implementor by collecting the style ids generically.

Reply:
Anyways, I'm not passionate about this. If this is the API you are happy with, than I'm ok with it, too.